devices/src/virtio/vsock/epoll_handler.rs

```diff
@@ -53,15 +76,65 @@ where
     /// Walk the driver-provided RX queue buffers and attempt to fill them up with any data that we
     /// have pending.
     ///
-    fn process_rx(&mut self) {
-        error!("vsock: EpollHandler::process_rx() NYI");
+    fn process_rx(&mut self) -> bool {
+        debug!("vsock: epoll_handler::process_rx()");
+
+        let mut have_used = false;
+
+        while let Some(head) = self.rxvq.pop(&self.mem) {
+            let used_len = match VsockPacket::from_rx_virtq_head(&head) {
+                Ok(mut pkt) => {
+                    if self.backend.recv_pkt(&mut pkt).is_ok() {
+                        pkt.hdr().len() as u32 + pkt.len()
+                    } else {
+                        // We are using a consuming iterator over the virtio buffers, so, if we can't
+                        // fill in this buffer, we'll need to undo the last iterator step.
+                        self.rxvq.undo_pop();
```

Comment:
If I understood correctly the purpose of the `have_used` variable, shouldn't it be false in case this happens?